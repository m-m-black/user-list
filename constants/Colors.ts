const tintColorLight = "#306FC7";
const tintColorDark = "#fff";

// These colours remain consistent across themes
export const avatarBackground = "#EAF2FA";
export const avatarText = "#306FC7";
export const avatarSubtitleText = "#808080";

export const Colors = {
  light: {
    text: "#11181C",
    background: "#fff",
    backgroundSecondary: "#EAF2FA",
    tint: tintColorLight,
    icon: "#687076",
    tabIconDefault: "#687076",
    tabIconSelected: tintColorLight,
  },
  dark: {
    text: "#ECEDEE",
    background: "#151718",
    backgroundSecondary: "#2F4F4F",
    tint: tintColorDark,
    icon: "#9BA1A6",
    tabIconDefault: "#9BA1A6",
    tabIconSelected: tintColorDark,
  },
};
