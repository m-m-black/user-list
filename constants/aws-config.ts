// These would normally be defined in a .env file
export const AWS_APPSYNC_URL =
  "https://prrwjjssnvhpbcdwbcwx3nm3zm.appsync-api.ap-southeast-2.amazonaws.com/graphql";
export const AWS_APPSYNC_API_KEY = "da2-6y6arb7mwvgrnmds2jignrgr2u";
