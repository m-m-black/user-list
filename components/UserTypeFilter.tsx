import { View, StyleSheet } from "react-native";
import { CheckBox } from "@rneui/themed";

import { UserRole } from "@/constants/UserRole";

type UserTypeFilterProps = {
  selectedRole: UserRole | undefined;
  setSelectedRole: (role: UserRole | undefined) => void;
};

export default function UserTypeFilter({
  selectedRole,
  setSelectedRole,
}: UserTypeFilterProps) {
  const toggleFilter = (role: UserRole) => {
    if (selectedRole === role) {
      setSelectedRole(undefined);
    } else {
      setSelectedRole(role);
    }
  };

  return (
    <View>
      <CheckBox
        checked={selectedRole === UserRole.ADMIN}
        onPress={() => toggleFilter(UserRole.ADMIN)}
        checkedIcon="dot-circle-o"
        uncheckedIcon="circle-o"
        title="Admin"
        containerStyle={[
          styles.container,
          selectedRole === UserRole.ADMIN && styles.containerSelected,
        ]}
        checkedColor="#306FC7"
      />
      <CheckBox
        checked={selectedRole === UserRole.MANAGER}
        onPress={() => toggleFilter(UserRole.MANAGER)}
        checkedIcon="dot-circle-o"
        uncheckedIcon="circle-o"
        title="Manager"
        containerStyle={[
          styles.container,
          selectedRole === UserRole.MANAGER && styles.containerSelected,
        ]}
        checkedColor="#306FC7"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 10,
  },
  containerSelected: {
    backgroundColor: "#EAF2FA",
  },
});
