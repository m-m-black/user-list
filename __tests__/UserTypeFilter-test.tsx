import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import UserTypeFilter from "@/components/UserTypeFilter";

import { UserRole } from "@/constants/UserRole";

// All components used from React Native Elements are mocked.
// See __mocks__/@rneui/themed.js
jest.mock("@rneui/themed");

describe("UserTypeFilter", () => {
  test("Admin radio button is selected when Admin role is selected", () => {
    const mockSetSelectedRole = jest.fn();

    const { getByText } = render(
      <UserTypeFilter
        selectedRole={UserRole.ADMIN}
        setSelectedRole={mockSetSelectedRole}
      />
    );

    // Confirm the Admin radio is selected
    const adminButton = getByText("⚫ Admin");
    expect(adminButton).toBeTruthy();

    // Confirm the Manager radio is unselected
    const managerButton = getByText("◯ Manager");
    expect(managerButton).toBeTruthy();
  });

  test("clicking Admin radio calls setSelectedRole with correct value", () => {
    const mockSetSelectedRole = jest.fn();

    const { getByText } = render(
      <UserTypeFilter
        selectedRole={undefined}
        setSelectedRole={mockSetSelectedRole}
      />
    );

    const adminButton = getByText("◯ Admin");
    fireEvent.press(adminButton);

    expect(mockSetSelectedRole).toHaveBeenCalledWith(UserRole.ADMIN);
  });

  test("clicking Admin radio when already selected clears selection", () => {
    const mockSetSelectedRole = jest.fn();

    const { getByText } = render(
      <UserTypeFilter
        selectedRole={UserRole.ADMIN}
        setSelectedRole={mockSetSelectedRole}
      />
    );

    const adminButton = getByText("⚫ Admin");
    fireEvent.press(adminButton);

    expect(mockSetSelectedRole).toHaveBeenCalledWith(undefined);
  });

  // Snapshot test to ensure UI remains consistent
  test("it renders correctly", () => {
    const mockSetSelectedRole = jest.fn();
    const tree = render(
      <UserTypeFilter
        selectedRole={UserRole.ADMIN}
        setSelectedRole={mockSetSelectedRole}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
