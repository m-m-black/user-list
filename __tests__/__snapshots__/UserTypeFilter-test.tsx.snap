// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTypeFilter it renders correctly 1`] = `
<View>
  <View
    checked={true}
    checkedColor="#306FC7"
    checkedIcon="dot-circle-o"
    containerStyle={
      [
        {
          "borderRadius": 10,
        },
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "#EAF2FA",
        },
      ]
    }
    onPress={[Function]}
    textStyle={
      {
        "color": "#11181C",
      }
    }
    title="Admin"
    uncheckedIcon="circle-o"
  >
    <Text>
      ⚫
       
      Admin
    </Text>
  </View>
  <View
    checked={false}
    checkedColor="#306FC7"
    checkedIcon="dot-circle-o"
    containerStyle={
      [
        {
          "borderRadius": 10,
        },
        {
          "backgroundColor": "#fff",
        },
        false,
      ]
    }
    onPress={[Function]}
    textStyle={
      {
        "color": "#11181C",
      }
    }
    title="Manager"
    uncheckedIcon="circle-o"
  >
    <Text>
      ◯
       
      Manager
    </Text>
  </View>
</View>
`;
